File: src/Java/504.ConvertToBase7.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.Queue;

class Solution {
    public String convertToBase7(int num) {
        if (num == 0) {
            return "0";
        }
        Integer.toString(num, 7);
        int absnum = num > 0 ? num : -num;
        int jinzhi = 7;
        LinkedList<Character> base7list = new LinkedList<>();
        while (absnum > 0){
            base7list.add((char)(absnum % jinzhi + '0'));
            absnum /= jinzhi;
        }       
        if (num < 0)
            base7list.add('-');
        Collections.reverse(base7list);
        return base7list.stream()
                .map(String::valueOf)
                .collect(Collectors.joining());
    }
}

File: src/Python/501-600/504.ConvertToBase7.py

class Solution:
    def convertToBase7(self, num):
        """
        :type num: int
        :rtype: str
        """
        if num == 0:
            return '0'
        base7list = []
        jinzhi = 7
        absnum = abs(num)
        while absnum > 0:
            base7list.append(str(absnum % jinzhi))
            absnum //= jinzhi
        if num < 0:
            base7list.append('-')
        return ''.join(base7list[::-1])
        
if __name__ == '__main__':
    solution = Solution()
    print(solution.convertToBase7(100))
    print(solution.convertToBase7(8))
    print(solution.convertToBase7(9))
    print(solution.convertToBase7(-7))
else:
    pass
